authors/apps/articles/serializers.py

```diff
@@ -66,3 +73,69 @@ class Meta:
         model = Bookmark
         fields = ['author', 'article_title', 'slug',
                   'description', 'bookmarked_at', 'image']
+
+
+class RatingSerializer(serializers.ModelSerializer):
+    """
+    class holding logic for article rating
+    """
+
+    article = serializers.PrimaryKeyRelatedField(
+        queryset=Article.objects.all())
+    rated_on = serializers.DateTimeField(read_only=True)
+    author = serializers.PrimaryKeyRelatedField(
+        queryset=User.objects.all())
+    score = serializers.DecimalField(required=True, max_digits=5,
+                                     decimal_places=2)
+
+    @staticmethod
+    def update_data(data, slug, user: User):
+        """
+        method to update the article with a rating
+        """
+        try:
+            article = Article.objects.get(slug__exact=slug)
+        except Article.DoesNotExist:
+            raise serializers.ValidationError("Article is not found.")
+
+        if article.author == user:
+            raise serializers.ValidationError({
+                "article": [
+                    "Please rate an article that does not belong to you"]
+            })
+
+        score = data.get("score", 0)
+        if score > 5 or score < 0:
+            raise serializers.ValidationError({
+                "score": ["Score value must not go "
+                          "below `0` and not go beyond `5`"]
+            })
+
+        data.update({"article": article.pk})
+        data.update({"author": user.pk})
+        return data
+
+    def create(self, validated_data):
+        """
+        method to create and save a rating for
+        """
+        author = validated_data.get("author", None)
+        article = validated_data.get("article", None)
+        score = validated_data.get("score", 0)
+
+        try:
```

Comment:
@akram256 , Can you override the get method to remove this `try`. Have a look how @NLSanyu does it.